src/SFA.DAS.FAT.Web/Controllers/ShortlistController.cs

```diff
@@ -15,139 +16,142 @@
+                Id = request.LarsCode,
+                ProviderId = request.Ukprn,
+                Added = string.IsNullOrEmpty(request.ProviderName) ? "" : WebEncoders.Base64UrlEncode(_protector.Protect(
+                    System.Text.Encoding.UTF8.GetBytes($"{request.ProviderName}")))
             });
+        }
 
-            if (!string.IsNullOrEmpty(request.RouteName))
-            {
-                return RedirectToRoute(request.RouteName, new
-                {
-                    Id = request.TrainingCode,
-                    ProviderId = request.Ukprn,
-                    Added = string.IsNullOrEmpty(request.ProviderName) ? "" : WebEncoders.Base64UrlEncode(_protector.Protect(
-                        System.Text.Encoding.UTF8.GetBytes($"{request.ProviderName}")))
-                });
-            }
+        return Accepted(result);
+    }
 
-            return Accepted(result);
-        }
+    [HttpPost]
+    [Route("items/{id}", Name = RouteNames.DeleteShortlistItem)]
+    public async Task<IActionResult> DeleteShortlistItemForUser(DeleteShortlistItemRequest request)
+    {
+        await _mediator.Send(new DeleteShortlistItemForUserCommand
+        {
+            Id = request.ShortlistId,
+        });
 
-        [HttpPost]
-        [Route("items/{id}", Name = RouteNames.DeleteShortlistItem)]
-        public async Task<IActionResult> DeleteShortlistItemForUser(DeleteShortlistItemRequest request)
+        if (!string.IsNullOrEmpty(request.RouteName))
         {
-            var cookie = _shortlistCookieService.Get(Constants.ShortlistCookieName);
-            if (cookie != null)
-            {
-                await _mediator.Send(new DeleteShortlistItemForUserCommand
-                {
-                    Id = request.ShortlistId,
-                    ShortlistUserId = cookie.ShortlistUserId
-                });
-            }
-            if (!string.IsNullOrEmpty(request.RouteName))
+            return RedirectToRoute(request.RouteName, new
             {
-                return RedirectToRoute(request.RouteName, new
-                {
-                    Id = request.TrainingCode,
-                    ProviderId = request.Ukprn,
-                    Removed = string.IsNullOrEmpty(request.ProviderName) ? "" : WebEncoders.Base64UrlEncode(_protector.Protect(
-                        System.Text.Encoding.UTF8.GetBytes($"{request.ProviderName}")))
-                });
-            }
-
-            return Accepted();
+                Id = request.TrainingCode,
+                ProviderId = request.Ukprn,
+                Removed = string.IsNullOrEmpty(request.ProviderName) ? "" : WebEncoders.Base64UrlEncode(_protector.Protect(
+                    System.Text.Encoding.UTF8.GetBytes($"{request.ProviderName}")))
+            });
         }
+
+        return Accepted();
+    }
+
+    /// <summary>
+    /// This is a workaround to update the shortlist component via java script
+    /// </summary>
```

Comment:
comment not needed imo

Reply:
The endpoint on its own when you look at it will not make any sense hence the comment.